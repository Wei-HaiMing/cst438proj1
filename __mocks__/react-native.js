// __mocks__/react-native.js
const React = require('react');

const StyleSheet = {
  create: (styles) => styles,
  flatten: (style) => style,
};

const Text = ({ children, ...props }) => React.createElement('Text', props, children);
const View = ({ children, ...props }) => React.createElement('View', props, children);

module.exports = {
  StyleSheet,
  Text,
  View,
};
