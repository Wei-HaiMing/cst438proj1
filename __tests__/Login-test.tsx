import Login from '@/app/login';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

const mockPush = jest.fn();
const mockReplace = jest.fn();
const mockGetAllAsync = jest.fn();
const mockExecAsync = jest.fn();


jest.mock('expo-router', () => ({
    useRouter: () => ({
        push: mockPush,
        replace: mockReplace,
    }),
}));

jest.mock("expo-sqlite", () => ({
    useSQLiteContext: () => ({
        getAllAsync: mockGetAllAsync,
        execAsync: mockExecAsync,
    }),
}));

describe("<Login />", () => {

    beforeEach(() => {
        jest.clearAllMocks();

        mockGetAllAsync.mockResolvedValue([]);
        mockExecAsync.mockResolvedValue(undefined);
    });

    it("renders login page correctly", () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
});