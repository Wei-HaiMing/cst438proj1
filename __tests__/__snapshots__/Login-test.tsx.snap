// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<Login /> renders login page correctly 1`] = `null`;
